### DISCUSSION TITLE
What's Frigate doing from midnight to 5am?

### DISCUSSION THREAD

**Discussion by paularmstrong** [blakeblackshear/frigate#366] [Ask A Question] [answered]
I'm using the Frigate beta, Glances addon, Influxdb and Grafana. Looking at the last 2 days, I'm noticing that Frigate's CPU usage drops dramatically from midnight to 5am, almost exactly. Is there some sort of scheduled thing happening that's making it way faster? Or maybe it's just not doing anything during that time period?

<img width="1207" alt="Screen Shot 2020-12-11 at 12 22 49 PM" src="https://user-images.githubusercontent.com/33297/101951310-a7776c00-3bab-11eb-8a80-ddf1523323d9.png">

**Comment by blakeblackshear** [comment, marked as answer]
There is nothing scheduled, but it's odd that its exact like that. Frigate does often use less CPU at night when there there is less moving in your frame. Part of the reason it is so efficient is that it does very little when there are no objects being tracked or motion to investigate.

I have seen cameras that do a scheduled nightly reboot. That reboot could cause frigate to stop tracking something it was actively watching before. The pickup in the morning is likely due to the switch from IR back to color mode.

**Reply by paularmstrong** [reply]
There should be nothing in frame on the cameras that I'm tracking (only person, cat, dog, and bicycle). It gets dark here by 4:30pm and won't be light until 7am, so I wouldn't expect the IR to affect it like it is at these times.

**Reply by blakeblackshear** [reply]
If you pull some of the values into grafana from the `/stats` endpoint, that will tell you more about what frigate is doing.

**Reply by paularmstrong** [reply]
Good idea! Pulled those up. Does anything here help?

<img width="1573" alt="Screen Shot 2020-12-11 at 1 10 18 PM" src="https://user-images.githubusercontent.com/33297/101955251-55861480-3bb2-11eb-93ac-ac087af91b7b.png">

**Reply by paularmstrong** [reply]
Seems like detection trickles out quite a bit before midnight (~22:00, as expected), then picks up after 06:00 just as I'm getting up (still dark out). The coral inference speed stabilizing matches up with the detection fps bumps as well.

**Reply by blakeblackshear** [reply]
It looks like there just isn't enough motion for frigate to run detection between those time frames. You could go look at the historical footage at those break points. I would guess something in the image changes when the detection fps values pick up in the morning.

**Reply by paularmstrong** [reply]
I actually think it might be that there's something else causing some bottleneck on my system. The following stats also go way up during this time
* iowait
* interrupts
* ctx_switches
* nic rx, tx

I'll need to look into my other containers to see what's spawning a bunch of processes... maybe Plex or something

**Reply by paularmstrong** [reply]
Welp, yep... Plex is the culprit. It's eating up all the network bandwidth 
<img width="513" alt="Screen Shot 2020-12-11 at 2 14 37 PM" src="https://user-images.githubusercontent.com/33297/101959982-43f53a80-3bbb-11eb-951c-34399b3757e7.png">

**Reply by RealKanashii** [reply]
@paularmstrong how you configured the glances docker to export to infludb? I'm interested in replicate your monitoring but Glances doesn`t allow "-w --export influxdb" config in "GLANCES_OPT" env.

**Reply by paularmstrong** [reply]
@RealKanashii First I set up a database and a user/password in the influxdb addon for it, then added the `influxdb` section to the glances config:

```yaml
influxdb:
  enabled: true
  host: a0d7b954-influxdb
  port: 8086
  username: REDACTED
  password: REDACTED
  database: glances
  prefix: localhost
  interval: 60
```

This is all using the addons for Home Assistant, of course
